src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -604,6 +604,15 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 					return m_polyMesh.subdivAdaptiveError == 0.0f || m_polyMesh.subdivAdaptiveSpace == g_objectSpace;
 				}
 			}
+			else if( const IECore::ExternalProcedural *procedural = IECore::runTimeCast<const IECore::ExternalProcedural>( object ) )
+			{
+				// We don't instance "ass archive" procedurals, because Arnold
+				// does automatic instancing of those itself, using its procedural
+				// cache.
+				return
```

Comment:
> Can we add () braces for clarity?

As in this?

```
return (
    !boost::ends_with( procedural->getFileName(), ".ass" ) &&
    !boost::ends_with( procedural->getFileName(), ".ass.gz" )
);
```

> Also, is this going to fix that light rig transform issue?

Ooh, I think it might. If I load a .ass with a light in an ExternalProcedural, and then transform that around during IPR, the light moves. That means the problem is fixed right?